pkg/aspect/query/query.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright © 2021 Aspect Build Systems Inc
+
+Not licensed for re-use.
+*/
+
+package query
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/manifoldco/promptui"
+	"github.com/spf13/cobra"
+
+	"aspect.build/cli/pkg/aspecterrors"
+	"aspect.build/cli/pkg/bazel"
+	"aspect.build/cli/pkg/ioutils"
+)
+
+type PresetQuery struct {
+	Name        string
+	Description string
+	Query       string
+}
+
+type Query struct {
+	ioutils.Streams
+
+	Bzl           bazel.Bazel
+	IsInteractive bool
+
+	Presets []*PresetQuery
+
+	GetAPrompt func(label string) PromptRunner
+	GetASelect func(presetNames []string) SelectRunner
```

Comment:
These function names were just an example. Not sure they are the most appropriate names. We can rename them on a follow-up PR.

Reply:
Maybe a more Go-idiomatic would be:

```suggestion
	Prompt func(label string) PromptRunner
	Select func(presetNames []string) SelectRunner
```